src/main/java/org/sigmah/client/ui/presenter/contact/ContactDetailsPresenter.java

```diff
@@ -578,78 +579,147 @@ public void onRequiredValueChange(RequiredValueEvent event) {
 
   private void updateContact(final ContactDTO contactDTO, final AsyncCallback<ContactDTO> callback, final Component target) {
 
-    // Checks if there are any changes regarding layout group iterations
-    dispatch.execute(new UpdateLayoutGroupIterations(new ArrayList<IterationChange>(iterationChanges.values()), contactDTO.getId()), new CommandResultHandler<ListResult<IterationChange>>() {
-
-      @Override
-      public void onCommandFailure(final Throwable caught) {
-        N10N.error(I18N.CONSTANTS.save(), I18N.CONSTANTS.saveError());
-      }
-
-      @Override
-      protected void onCommandSuccess(ListResult<IterationChange> result) {
-
-        for (IterationChange iterationChange : result.getList()) {
-          if (iterationChange.isDeleted()) {
-            // remove corresponding valueEvents
-
-            Iterator<ValueEvent> valuesIterator = valueChanges.iterator();
-            while (valuesIterator.hasNext()) {
-              ValueEvent valueEvent = valuesIterator.next();
-
-              if (valueEvent.getIterationId() == iterationChange.getIterationId()) {
-                valuesIterator.remove();
-              }
-            }
-          } else if (iterationChange.isCreated()) {
-            // change ids in valueEvents
-            int oldId = iterationChange.getIterationId();
-            int newId = iterationChange.getNewIterationId();
-
-            // updating tabitem id
-            newIterationsTabItems.get(oldId).setIterationId(newId);
-
-            for (ValueEvent valueEvent : valueChanges) {
-              if (valueEvent.getIterationId() == oldId) {
-                valueEvent.setIterationId(newId);
-              }
-            }
-          }
-        }
-
-        iterationChanges.clear();
-        newIterationsTabItems.clear();
-
-        dispatch.execute(new UpdateContact(contactDTO.getId(), valueChanges), new CommandResultHandler<VoidResult>() {
-
-          @Override
-          public void onCommandFailure(final Throwable caught) {
-            N10N.error(I18N.CONSTANTS.save(), I18N.CONSTANTS.saveError());
-          }
-
-          @Override
-          protected void onCommandSuccess(final VoidResult result) {
-
-            N10N.infoNotif(I18N.CONSTANTS.infoConfirmation(), I18N.CONSTANTS.saveConfirm());
-            eventBus.fireEvent(new UpdateEvent(UpdateEvent.CONTACT_UPDATE, contactDTO));
-
-            // Checks if there is any update needed to the local project instance.
-            for (final ValueEvent event : valueChanges) {
-              if (event.getSource() instanceof DefaultContactFlexibleElementDTO) {
-                updateCurrentContact(contactDTO, (DefaultContactFlexibleElementDTO) event.getSource(), event.getSingleValue());
-              }
-            }
+		if (!iterationChanges.isEmpty()) {
```

Comment:
Too long, please break in smaller methods (or maybe extract anonymous class to inner?)